service/host.go

```diff
@@ -223,20 +225,33 @@ func (s *host) Start(waitForShutdown bool) (<-chan Exit, <-chan struct{}, error)
 				if _, err := s.shutdown(e, "", nil); err != nil {
 					ulog.Logger().Error("Unable to shut down modules", "initialError", e, "shutdownError", err)
 				}
-				s.shutdownMu.Lock()
-
 				return errChan, readyCh, e
 			}
 		}
 	}
 
+	s.shutdownMu.Unlock()
```

Comment:
Changed this around to avoid a deadlock. The mutex was only there to guard `s.locked` so it's fine to unlock here.